src/views/examples/Register.js

```diff
@@ -1,166 +1,181 @@
 /*!
-
 =========================================================
 * Argon Dashboard React - v1.2.0
 =========================================================
-
 * Product Page: https://www.creative-tim.com/product/argon-dashboard-react
 * Copyright 2021 Creative Tim (https://www.creative-tim.com)
 * Licensed under MIT (https://github.com/creativetimofficial/argon-dashboard-react/blob/master/LICENSE.md)
-
 * Coded by Creative Tim
-
 =========================================================
-
 * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
-
 */
-import React from "react";
+import React, {useState } from "react";
+import { useDispatch } from "react-redux";
+import { register } from "../../actions/userActions";
+import { useHistory } from "react-router";
 
-// reactstrap components
 import {
   Button,
   Card,
-  CardHeader,
   CardBody,
   FormGroup,
   Form,
   Input,
   InputGroupAddon,
   InputGroupText,
   InputGroup,
-  Row,
   Col,
 } from "reactstrap";
 
-const Register = () => {
-  return (
+
+const Register=()=> {
+  
+  const [password, setPassword]= useState('');
+  const [confirmpassword, setConfirmpassword]=useState('');
+  const [pass, passState]=useState('');
+  const [email, setEmail] = useState("");
+  const [emailError, setEmailError] = useState('')
+  const [isDisabled, setIsdisabled]= useState(true)
+  const history=useHistory();
+
+  const validateEmail = async (e) => {
+    var class_valid = "is-valid form-control";
+    var class_invalid = "is-invalid form-control";
+    
+    if (
+      /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@iitk.ac.in/.test(e.value)
+    ){
+      setEmail(e.value);
+      setEmailError("  ");
+      e.className=class_valid;
+    }
+    else{
+      setEmailError("invalid iitk email");
+      e.className=class_invalid;
+    }
+  }
+  
+  const onRegister = async (e)=> {
+   
+    if (e) e.preventDefault();
+    try{
+      const user ={
+        email: email,
+        password: password
+      };
+      await register(user);
+      //await dispatch(register(user));
+      history.push("../../../");
```

Comment:
Please use dispatch here and ignore using such type of navigation. If user is successfully saved it automatically redirect to Admin page no need for some routing here. You can refer how login part is done.